sound/soc/sof/topology.c

```diff
@@ -951,13 +921,35 @@ static int sof_parse_tokens(struct snd_soc_component *scomp,
 		array = (struct snd_soc_tplg_vendor_array *)((u8 *)array
 			+ asize);
 
+		/* update parse position in destination */
+		if (found && !(found % count))
+			offset += objectsize;
```

Comment:
@juimonen This seems correct, but took me a while to parse. I wonder the code would be more readable if you just had a "int set;" to track which set you are tracking and reset "found = 0;" after each set. 

The you had:

if found <= count) { ++set; found = 0; offset += objectsize; } 

And:

while (found*set <= sets*count)

... oh well, @ranj063 and @lyakh feel free to chime in. You didn't comment on this, so maybe this is just me. :)

Reply:
It didn't strike me as particularly complex

Reply:
any code that forces reviewers to think is usually unmaintainable. please simplify or add comments. There's some alignment here that should be made explicit.

Reply:
in "Finnish" it reads "update offset when found is multiple of count, but not 0", so as a set has "count" values, always when we reach multiple of "count" we need to move to new destination struct by offset. But yes it is 2 against 1 reviewers, so will change to something more readable then...

Reply:
looks fine to me too. Maybe a comment saying that offset needs to be incremented when one set of tokens have been found

Reply:
ok

Reply:
ok I needed to change this part as it doesn't take into account that the similar sets might have other arrays in between. So it assumed that the sets follow each other and the offset is also updated in that case (which is wrong).